coverage/sqldata.py

```diff
@@ -182,8 +182,16 @@ def _file_id(self, filename, add=False):
         if filename not in self._file_map:
             if add:
                 with self._connect() as con:
-                    cur = con.execute("insert into file (path) values (?)", (filename,))
-                    self._file_map[filename] = cur.lastrowid
+                    try:
+                        cur = con.execute("insert into file (path) values (?)", (filename,))
```

Comment:
how about skipping the extra query and handling the duplication in SQL?

```suggestion
                        cur = con.execute("insert OR REPLACE into file (path) values (?)", (filename,))
```

i'm not sure how / if this will change the row id however, but it feels more elegant than handling the exception which is a little brittle: what if the error message changes? what if the column name changes?

Reply:
Makes sense.
Can adjust if @nedbat agrees / wants to handle it this way after all.

Reply:
Changed to "insert or replace", thanks!